### PR TITLE
Adding ephemeralstorage and ResourceStorage for kaniko pods

### PR DESCRIPTION
**Pull request by prary**
#2990  Adding ephemeral storage for kaniko pod

**Before**

```
requests: 
  cpu:             2
  memory:             1Gi
limits: 
  cpu:             2
  memory:             1Gi
```
**After**
```
requests: 
  cpu:             2
  memory:             1Gi
  ephemeralStorage : 1Gi
limits: 
  cpu:             2
  memory:             1Gi
  ephemeralStorage : 1Gi
```
**Next PRs.**
N/A

### CODE REVIEW

File: pkg/skaffold/build/cluster/sources/sources.go [hunk 1]
```diff
@@ -184,6 +188,9 @@ func resourceRequirements(rr *latest.ResourceRequirements) v1.ResourceRequiremen
 			if rr.Requests.Memory != "" {
 				req.Requests[v1.ResourceMemory] = resource.MustParse(rr.Requests.Memory)
 			}
+			if rr.Requests.EphemeralStorage != "" {
```
Comment (tejal29) [inline pkg/skaffold/build/cluster/sources/sources.go]: please add tests in `source_test.go`

**Comment by prary** [comment]
hi @dgageot I will add test case and yes it totally make sense configuring ResourceStorage as well.
Just to confirm test case testing resourceRequirements function right?

**Comment by dgageot** [comment]
@prary yes. Thanks a lot!

File: pkg/skaffold/build/cluster/sources/sources.go (hunk 1 above)
Comment (prary) [reply pkg/skaffold/build/cluster/sources/sources.go]: added

**Comment by prary** [comment]
Not sure why build is failing it is passing locally and I did ran make generate - schema

**Comment by dgageot** [comment]
@prary This happens on one of my PRs too. Not sure what's happening...

**Comment by balopat** [comment]
My guess is that this happens when there is a schema change in master. Can you try to rebase and try locally?

**Comment by dgageot** [comment]
@prary It looks like you changed latest schema that's v17 now, not v16. I had the same issue and `make generate-schemas` didn't fix the issue. I need to investigate

**Pull request merged**